src/components/layout/Main.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from "react";
+import ProductCard from "../productsCard/ProductCard";
+import { fetchData } from "../../utils/fetch-data";
+
+
+
+interface Product {
+  id: number;
+  title: string;
+  artist: string;
+  price: number;
+  image_url: string;
+}
+
+const Main = () => {
+  const [productData, setProductData] = useState<Product[]>([]);
+  const [cartCount, setCartCount] = useState(0);
+
+  useEffect(() => {
+    const fetchProductData = async () => {
+      try {
+        const data: Product[] = await fetchData("vinyls", "GET");
+        setProductData(data);
+      } catch (error) {
+        console.error("Error fetching product data:", error);
+      }
+    };
+
+    fetchProductData();
+  }, []);
+
+  const addToCart = () => {
+    setCartCount((prevCount) => prevCount + 1);
+  };
+
+  return (
+    <main
```

Comment:
We now have 2 mains (with the main in App.tsx), so this should probably be a div if we need it - or could we move the styling into App.tsx or vice-versa?